packages/aws_common/lib/src/logging/aws_logger.dart

```diff
@@ -72,10 +87,38 @@ class AWSLogger implements Closeable {
     return AWSLogger('$namespace.$name');
   }
 
+  /// Returns a plugin of type [Plugin] registered to this
+  /// logger hierarchy or null.
+  Plugin? getPlugin<Plugin extends AWSLoggerPlugin>() {
+    var registeredPlugin = parent?.getPlugin<Plugin>();
+    return registeredPlugin ??= _subscriptions.keys
+        .firstWhereOrNull((element) => element is Plugin) as Plugin?;
+  }
+
   /// Registers an [AWSLoggerPlugin] to handle logs emitted by this logger
   /// instance.
-  void registerPlugin(AWSLoggerPlugin plugin) {
-    unregisterPlugin(plugin);
+  ///
+  /// Throws [StateError] if a plugin with same type is registered to this
+  /// logger hierarchy.
+  /// Unregisteres plugins with same type from all children in the hierarchy
+  /// to avoid duplicate logging.
```

Comment:
```suggestion
  /// Throws [StateError] if a plugin with same type is registered to this
  /// logger hierarchy.
  ///
  /// Unregisters plugins with same type from all children in the hierarchy
  /// to avoid duplicate logging.
```